account_analytic_required/models/account.py

```diff
@@ -85,27 +67,23 @@ def _check_analytic_required_msg(self):
                 "account": self.account_id.display_name,
                 "move": self.name or "",
             }
-        elif analytic_policy == "never" and (
-            self.analytic_account_id or self._has_analytic_distribution()
-        ):
-            analytic_account = (
-                self.analytic_account_id
-                or self.analytic_tag_ids.analytic_distribution_ids[:1]
-            )
+        elif analytic_policy == "never" and (self.analytic_distribution):
+            analytic_account = self.analytic_distribution
+            aaacount_ids = [int(k) for k in analytic_account.keys()]
+            aaacounts = self.env["account.analytic.account"].browse(aaacount_ids)
```

Comment:
maybe better `analytic_acc_ids` and `analytic_accs` to shorten the vars names

Reply:
updated !